build.xml

```diff
@@ -90,7 +90,10 @@
     <property name="test.jvm.args" value="" />
     <property name="testtag.extra" value="" />
     <property name="dist.dir" value="${build.dir}/dist"/>
-    <property name="tmp.dir" value="${java.io.tmpdir}"/>
+
+    <!-- Use build/tmp for temp files if not otherwise specified. Because Ant properties are immutable, this has no effect if
+         the user specifies the tmp.dir property -->
+    <property name="tmp.dir" value="${build.dir}/tmp"/>
```

Comment:
Does this conflict, override, etc. [this](https://github.com/apache/cassandra/blob/97d23f40c9c7c3485b453fad5030f6427db5ca46/build.xml#L1127). Basically how they both play together?

Reply:
Basically, the property defines what is used for the `java.io.tmpdir` property in testing. Ant properties are immutable, so if someone provides the property on the command line (e.g. `ant -Dtmp.dir=/tmp/mytests ...`) then line 96 has no effect and the tests will use the provided directory. Now that you've mentioned it, though, I should make a change so that this honors the previous definition, which would use any provided `java.io.tmpdir` property on the command line. I can make it conditional so that we'll still use `${build.dir}/tmp` if `java.io.tmpdir` is not provided

Reply:
Actually, that won't work since we're running inside `ant` and the property will always be set since it's the JVM. I can't think of a way to differentiate between a JVM default `java.io.tmpdir` and one that was passed on the ant command line...